pkg/tests/api/alerting/api_convert_prometheus_test.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/grafana/grafana/pkg/services/org"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/grafana/grafana/pkg/tests/testinfra"
+)
+
+func TestIntegrationConvertPrometheusEndpoints(t *testing.T) {
+	testinfra.SQLiteIntegrationTest(t)
+
+	// Setup Grafana and its Database
+	dir, path := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+		DisableLegacyAlerting: true,
+		EnableUnifiedAlerting: true,
+		DisableAnonymous:      true,
+		AppModeProduction:     true,
+		EnableFeatureToggles:  []string{"alertingConversionAPI"},
+	})
+
+	grafanaListedAddr, env := testinfra.StartGrafanaEnv(t, dir, path)
+
+	// Create a user to make authenticated requests
+	createUser(t, env.SQLStore, env.Cfg, user.CreateUserCommand{
+		DefaultOrgRole: string(org.RoleAdmin),
+		Password:       "password",
+		Login:          "admin",
+	})
+
+	apiClient := newAlertingApiClient(grafanaListedAddr, "admin", "password")
+
+	namespace := "test-namespace"
+
+	durationPtr := func(d time.Duration) *prommodel.Duration {
+		dur := prommodel.Duration(d)
+		return &dur
+	}
+
+	promGroup1 := apimodels.PrometheusRuleGroup{
+		Name:     "test-group-1",
+		Interval: prommodel.Duration(60 * time.Second),
+		Rules: []apimodels.PrometheusRule{
+			// Recording rule
+			{
+				Record: "test:requests:rate5m",
+				Expr:   "sum(rate(test_requests_total[5m])) by (job)",
+				Labels: map[string]string{
+					"env":  "prod",
+					"team": "infra",
+				},
+			},
+			// Two alerting rules
+			{
+				Alert: "HighMemoryUsage",
+				Expr:  "process_memory_usage > 80",
+				For:   durationPtr(5 * time.Minute),
+				Labels: map[string]string{
+					"severity": "warning",
+					"team":     "alerting",
+				},
+				Annotations: map[string]string{
+					"annotation-1": "value-1",
+					"annotation-2": "value-2",
+				},
+			},
+			{
+				Alert: "ServiceDown",
+				Expr:  "up == 0",
+				For:   durationPtr(2 * time.Minute),
+				Labels: map[string]string{
+					"severity": "critical",
+				},
+				Annotations: map[string]string{
+					"annotation-1": "value-1",
+				},
+			},
+		},
+	}
+
+	promGroup2 := apimodels.PrometheusRuleGroup{
+		Name:     "test-group-2",
+		Interval: prommodel.Duration(60 * time.Second),
+		Rules: []apimodels.PrometheusRule{
+			{
+				Alert: "HighDiskUsage",
+				Expr:  "disk_usage > 80",
+				For:   durationPtr(2 * time.Minute),
+				Labels: map[string]string{
+					"severity": "low",
+					"team":     "alerting",
+				},
+				Annotations: map[string]string{
+					"annotation-5": "value-5",
+				},
+			},
+		},
+	}
+
+	ds := apiClient.CreateDatasource(t, datasources.DS_PROMETHEUS)
+
+	t.Run("create two rule groups and get them back", func(t *testing.T) {
+		apiClient.ConvertPrometheusPostRuleGroup(t, namespace, ds.Body.Datasource.UID, promGroup1)
+		apiClient.ConvertPrometheusPostRuleGroup(t, namespace, ds.Body.Datasource.UID, promGroup2)
+
+		ns, _, _ := apiClient.GetAllRulesWithStatus(t)
```

Comment:
When the reading API starts working, I'll replace this with reading and comparing the result structures, but for now we check just the number of returned rules.